tools/report-converter/codechecker_report_converter/smatch/output_parser.py

```diff
@@ -0,0 +1,60 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+
+import logging
+import os
+import re
+
+from ..output_parser import BaseParser, Message
+LOG = logging.getLogger('ReportConverter')
+
+
+class SmatchParser(BaseParser):
+    """
+    Parser for Smatch Output
+    """
+
+    def __init__(self, analyzer_result):
+        super(SmatchParser, self).__init__()
+
+        self.analyzer_result = analyzer_result
+
+        self.message_line_re = re.compile(
+            # File path followed by a ':'.
+            r'^(?P<path>[\S ]+?):'
+            # Line number followed by a whitespace.
+            r'(?P<line>\d+?)\s'
+            # Message.
+            r'(?P<message>[\S \t]+)\s*')
+
+    def parse_message(self, it, line):
+        """
+        Actual Parsing function for the given line
```

Comment:
please highlight that it is expected that each line contains a separate report (there are no multi line reports)

Reply:
Yes, Will put that into the comment